packages/components/src/border-box-control/test/index.js

```diff
@@ -210,27 +235,52 @@ describe( 'BorderBoxControl', () => {
 			expect( widthInputs[ 3 ].value ).toBe( '1' ); // Bottom.
 		} );
 
-		it( 'should omit style options when requested', () => {
-			renderBorderBoxControl( { enableStyle: false } );
-			clickButton( 'Unlink sides' );
+		it( 'should omit style options when requested', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
+			} );
 
-			const colorButtons = screen.getAllByLabelText( colorPickerRegex );
+			render( <BorderBoxControl { ...props } enableStyle={ false } /> );
 
-			colorButtons.forEach( ( button ) => {
-				fireEvent.click( button );
+			await user.click(
+				screen.getByRole( 'button', { name: 'Unlink sides' } )
+			);
+
+			const colorButtons = screen.getAllByLabelText( colorPickerRegex );
 
+			function assertStyleOptionsMissing() {
 				const styleLabel = screen.queryByText( 'Style' );
-				const solidButton = queryButton( 'Solid' );
-				const dashedButton = queryButton( 'Dashed' );
-				const dottedButton = queryButton( 'Dotted' );
+				const solidButton = screen.queryByRole( 'button', {
+					name: 'Solid',
+				} );
+				const dashedButton = screen.queryByRole( 'button', {
+					name: 'Dashed',
+				} );
+				const dottedButton = screen.queryByRole( 'button', {
+					name: 'Dotted',
+				} );
 
 				expect( styleLabel ).not.toBeInTheDocument();
 				expect( solidButton ).not.toBeInTheDocument();
 				expect( dashedButton ).not.toBeInTheDocument();
 				expect( dottedButton ).not.toBeInTheDocument();
+			}
 
-				fireEvent.click( button );
-			} );
+			await user.click( colorButtons[ 0 ] );
+			assertStyleOptionsMissing();
+			await user.click( colorButtons[ 0 ] );
+
+			await user.click( colorButtons[ 1 ] );
+			assertStyleOptionsMissing();
+			await user.click( colorButtons[ 1 ] );
+
+			await user.click( colorButtons[ 2 ] );
+			assertStyleOptionsMissing();
+			await user.click( colorButtons[ 2 ] );
+
+			await user.click( colorButtons[ 3 ] );
+			assertStyleOptionsMissing();
+			await user.click( colorButtons[ 3 ] );
```

Comment:
It looks like this test is timing out — could it be somehow related to the refactor from `forEach` to a linear set of instructions?

Reply:
I think that's the reason - the user-event clicks are much more complicated than just simulating a single event. And because there's quite a few here, plus quite a few assertions between every click pair, this could take longer on some setups. For example, I never got to fail it locally. I've increased the timeout in https://github.com/WordPress/gutenberg/pull/45208/commits/4e6c822da787a3327c1867b83335704e6e5f266d to fix this, but since we're working with fake timers, it doesn't make the test slower, so we're good.